docs/README.md

```diff
@@ -27,6 +26,63 @@ type: docs
 * inclusion of `type: docs` is important for the theme to properly index the site contents
 * the `linkTitle` attribute allows you to simplify the name as it appears in the left-side nav bar - ideally it should be short and clear - whereas the title can handle longform names for pages/documents.
 
+## Links
+
+### External Links
+
+Links to anything outside of the `docs/content/` directory should be considered an external link - when the website is built, any relative paths to other files and sections in the OSM repo that were navigable within Github will likely break when clicked from the compiled pages at docs.openservicemesh.io.
+
+To ensure urls work in both settings, please use an absolute path when referencing resources outside of the website content structure.
+
+Example:
+
+```
+Take a look at [the following test](../pkg/configurator/client_test.go)
+```
+
+Should use the a full, absolute path:
+
+```
+Take a look at [the following test](https://github.com/openservicemesh/osm/blob/release-v0.2/pkg/configurator/client_test.go)
```

Comment:
We need some guidance/script on updating URLs with every release. Is this captured in the readme?

Reply:
> Note that the full path **can** also specify the release

I've worded the Readme in this PR to say `can` rather than `should`... 

My recommendation would be **to not do this** in the current documentation (aka on `main`). It's an extra step that would require regular edits to bump the version and I think would be an annoyance, so I suggest leaving `main` alone and instead to look at including version info later on where it becomes useful:

## Value of Versioned URLs

The advantage of versioned urls is to preserve external links - so that when a doc points to a different part of the OSM repo, it can be frozen in time to avoid causing confusion if that destination changes / moves / is deleted in subsequent releases. Versioned urls do nothing for the current, latest docs except add more maintenance overhead IMO. 

## Set a Version When Cutting a Release

I think versioning makes more sense when a new release is cut, say v0.8; that is when the [Process to Version the Docs Site](https://github.com/openservicemesh/osm/tree/main/docs#versioning-the-docs-site) would come into play.

At that point:

```
0.8 is now prod, aka docs.openservicemesh.io 
0.7 docs are spun off as an archive at https://release-v0-7--osm-docs.netlify.app
```

This is where versioned URLs become useful IMO - if someone is browsing the `release-v0-7` docs because they are still running that version, it's very likely they will hit broken links for anything that has changed on the live site. At the point the v0.7 docs are archival, so that's when urls should be updated:

```
example:

// the docs link to the helm chart as follows:
https://github.com/openservicemesh/osm/blob/main/charts/osm/Chart.yaml

// when an archival doc is being created, the ideal path would be 
https://github.com/openservicemesh/osm/blob/release-v0.7/charts/osm/Chart.yaml
```

i.e. add a commit to the `release-v0-7` branch which rewrites the urls as above.

---

This is just my opinion - what do you think? 

It's a workflow that needs input to find the cleanest process for the team. I plan to help with this process for the next release, so I can run through this and make sure it's carefully documented with the team.

Reply:
Thanks for the detailed info @flynnduism.